src/core/public/saved_objects/saved_objects_client.ts

```diff
@@ -421,6 +425,26 @@ export class SavedObjectsClient {
     return request;
   }
 
+  /**
+   * Resolves a single object
+   *
+   * @param {string} type
+   * @param {string} id
+   * @returns The resolve result for the saved object for the given type and id.
+   */
+  public resolve = <T = unknown>(type: string, id: string) => {
+    if (!type || !id) {
+      return Promise.reject(new Error('requires type and id'));
+    }
+
+    const path = `${this.getPath(['resolve'])}/${type}/${id}`;
+    const request: Promise<SavedObjectsResolveResponse<T>> = this.savedObjectsFetch(path, {});
+    return request.then(({ saved_object: object, outcome, aliasTargetId }) => {
+      const savedObject = new SimpleSavedObject<T>(this, cloneDeep(object));
+      return new ResolvedSimpleSavedObject(savedObject, outcome, aliasTargetId);
+    });
+  };
```

Comment:
I started to try to change this but it affects every other function in the class, because `http.fetch` normally returns an `any` type. I figure we don't want to use an explicit `any`:
```
private savedObjectsFetch<TResponse = any>
```
so if it's OK with you, I'll leave this as-is and the whole class can be refactored in the future :+1:

Reply:
Yea, sounds good.